include/adc.h

```diff
@@ -253,6 +253,16 @@ struct adc_sequence {
 	 * a specific mode (e.g. when sampling multiple channels).
 	 */
 	u8_t oversampling;
+
+	/**
+	 * Optionally perform calibration before the reading is taken.
+	 *
+	 * The impact of channel configuration on the calibration
+	 * process is specific to the underlying hardware.  ADC
+	 * implementations that do not support calibration may ignore
+	 * this flag.
```

Comment:
My intent was to acknowledge that Zephyr's architectural assumption that all devices for a given subsystem have the same set of capabilities runs into problems with calibration and SAADC, without trying to document why.

We certainly could reject any attempt to request calibration when the hardware doesn't support it.  That could be annoying in "portable" code that wants to do calibration where it's supported.

Also, I believe that whether calibration is performed depends on how the SAADC channels are set up, viz. its behavior is dependent on the configuration of CH[0].  So people can't just assume that because it was requested, and the driver didn't return an error, that it actually happened.

We can leave this as is, or change "...may ignore this flag" to "should return -ENOTSUP if this flag is set".  I really don't care, but I don't want to change it more than one more time in this PR.

Reply:
> We can leave this as is, or change "...may ignore this flag" to "should return -ENOTSUP if this flag is set". I really don't care, but I don't want to change it more than one more time in this PR.

I thought about stating it rather as "... should ignore this flag", exactly to make possible to create such "portable" code that you mentioned. And I think it would be good to also add this flag (set to `true`) to the [basic test case](https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/drivers/adc/adc_api/src/test_adc.c#L251) for the ADC driver, to additionally show that drivers are expected to not return an error when they don't support calibration.